appenders/python/pulse/metrics.py

```diff
@@ -0,0 +1,167 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+# Including Apache 2 license because unix_time_micros method was pulled
+# from Apache Kudu project.
+
+from datetime import datetime
+import atexit
+
+from pytz import utc
+import six
+
+from pulse.batcher import PulseBatcher
+
+
+# noinspection PyPep8Naming
+class MetricWriter(PulseBatcher):
+    """
+    A class which sends metrics to the Pulse Log Collector.
+    """
+
+    def __init__(self, endpoint, table, capacity=1000,
+                 threadCount=1, logger=None):
+        """
+        Initializes a MetricWriter using the provided endpoint, table and
+        optional buffer capacity.
+
+        :param endpoint: The REST API endpoint for the Pulse Log Collector
+        :type endpoint: str
+        :param table: Kudu table to store metrics in
+        :type table: str
+        :param capacity: The maximum number of metrics to buffer before
+                                flushing. If no value is provided, each metric
+                                will be submitted one at a time. Defaults to
+                                1000.
+        :param threadCount: Number of threads to handle post requests.
+        :type threadCount: int
+        :param logger: :class:`logging.Logger` object for debug logging. If
+                       none is provided, a StreamHandler will be created to log
+                       to sys.stdout. It is recommended that you provide a
+                       logger if you plan to use more than one instance of this
+                       class.
+        :type logger: logging.Logger
+        :rtype: MetricWriter
+        """
+        PulseBatcher.__init__(self, endpoint, capacity, threadCount, logger)
+        self.table = table
+        # Cleanup when Python terminates
+        atexit.register(self.close)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.close()
+
+    def gaugeTimestamp(self, key, tag, value, timestamp,
```

Comment:
Should I change the 'id' column to 'key'? Was thinking 'ns' (namespace) might work alternatively.
key, tag, value
key, metric, value
ns, metric, value
id, metric, value

I don't feel very strongly about any of them. Would be good to have consistency with the table though.